figo/HttpsRequest.php

```diff
@@ -84,37 +105,64 @@ public function request($path, $data, $method, array $headers) {
+        $loggingData = array(
+            'path' => $path,
+            'data' => $data,
+            'method' => $method,
+            'headers' => $headers,
+            'response' => array(
+                'status' => $code,
+                'body' => $responseArray
+            )
+        );
+
+        if (strpos($path, '/task/progress') !== false) {
+            // when on /task/progress
+            if ($code >= 200 && $code < 400) {
+                $data = array_merge($loggingData, array('task_id' => explode('/task/progress?id=', $path)[1]));
+                if($responseArray['is_erroneous']) {
+                    $this->logger->info('API request to /task/progress', $data);
+                } else {
+                    $this->logger->debug('API request to /task/progress', $data);
+                }
+            } else {
+                $this->logger->info('API request to /task/progress failed', $loggingData);
+            }
+        } else {
+            // when not on /task/progress
+            if ($code >= 200 && $code < 400) {
+                $this->logger->debug('API request', $loggingData);
+            } else {
+                $this->logger->info('API request failed', $loggingData);
+            }
+        }
+
+
         // Evaluate HTTP response.
         if ($code >= 200 && $code < 300) {
             if (strlen($body) === 0) {
                 return array();
-            } else {
-                $obj = json_decode($body, true);
-                if (is_null($obj)) {
-                    throw new Exception("json_error", "Cannot decode JSON object.");
-                } else {
-                    return $obj;
-                }
-             }
-        } elseif ($code === 400) {
-             $err = json_decode($body, true);
-             if (is_null($err)) {
-                 throw new Exception("json_error", "Cannot decode JSON object.");
-             } else {
-                 throw new Exception($err["error"]["name"] .":". $err["error"]["message"]." (Error-Code: ".$err["error"]["code"].")" , $err["error"]["description"]);
-             }
-        } elseif ($code === 401) {
-            throw new Exception("unauthorized", "Missing, invalid or expired access token.");
-        } elseif ($code === 403) {
-            throw new Exception("forbidden", "Insufficient permission.");
-        } elseif ($code === 404) {
-            return null;
-        } elseif ($code === 405) {
-            throw new Exception("method_not_allowed", "Unexpected request method.");
-        } elseif ($code === 503) {
-            throw new Exception("service_unavailable", "Exceeded rate limit.");
+            }
+        }
+
+        $obj = json_decode($body, true);
+        if (is_null($obj)) {
+            throw new Exception("json_error", "Cannot decode JSON object.");
         } else {
-            throw new Exception("internal_server_error", "We are very sorry, but something went wrong.");
+            if ($code >= 200 && $code < 300) {
```

Comment:
you are checking all the status codes twice you might want to add the logging in here.